app/forms/provider_interface/rejections_wizard.rb

```diff
@@ -18,6 +18,24 @@ def rejection_reasons
 
     attr_accessor(*attribute_names)
 
+    def sanitize_attrs(attrs)
+      if attrs[:current_step] == 'new' && attrs.key?(:selected_reasons)
+        self.class.reasons.each do |reason|
+          if attrs[:selected_reasons].exclude?(reason.id)
+            attrs[reason.selected_reasons_attr_name] = [] if reason.reasons
+            attrs[reason.details.id] = nil if reason.details
+          end
+
+          unselected_nested_reasons = reason.reasons&.select { |nr| attrs[reason.selected_reasons_attr_name].exclude?(nr.id) }
```

Comment:
as discussed might be worth refactoring and leveraging recursion and breaking this out into multiple methods?